app/frontend/actions/common.ts

```diff
@@ -46,10 +46,14 @@ export default (store: Store) => {
     })
   }
 
-  const prepareTxPlan = async (args: TxPlanArgs): Promise<TxPlanResult> => {
+  const prepareTxPlan = async (args: TxPlanArgs): Promise<TxPlanResult | undefined> => {
     const state = getState()
+    const wallet = await getWallet()
     try {
-      return await getWallet()
+      if (!wallet) {
+        throw new Error('Wallet is not loaded')
```

Comment:
I would rather extract this logic into a `getWalletOrThrow` helper function which would be guaranteed to return the wallet, making the places where we invoke it less cluttered with checks that shouldn't really trigger at runtime unless there's a bug